packages/typespec-ts/test/azureIntegration/azureArmResources.spec.ts

```diff
@@ -454,4 +544,478 @@ describe("Azure Arm Resources Rest Client", () => {
+        allowReserved: true
+      },
+      "extension"
+    )
+      .get();
+    assert.strictEqual(resourceGroupResult.status, "200");
+    assert.deepEqual(resourceGroupResult.body, validResourceGroupExtensionsResource);
+
+    const subscriptionResult = await client.path(
+      "/{resourceUri}/providers/Azure.ResourceManager.Resources/extensionsResources/{extensionsResourceName}",
+      {
+        value: `subscriptions/${SUBSCRIPTION_ID_EXPECTED}`,
+        allowReserved: true
+      },
+      "extension"
+    )
+      .get();
+    assert.strictEqual(subscriptionResult.status, "200");
+    assert.deepEqual(subscriptionResult.body, validSubscriptionExtensionsResource);
+
+    const tenantResult = await client.path(
+      "/{resourceUri}/providers/Azure.ResourceManager.Resources/extensionsResources/{extensionsResourceName}",
+      {
+        value: "",
+        allowReserved: true
+      },
+      "extension"
+    )
+      .get();
+    assert.strictEqual(tenantResult.status, "200");
+    assert.deepEqual(tenantResult.body, validTenantExtensionsResource);
+
+    const resourceResult = await client.path(
+      "/{resourceUri}/providers/Azure.ResourceManager.Resources/extensionsResources/{extensionsResourceName}",
+      {
+        value: `subscriptions/${SUBSCRIPTION_ID_EXPECTED}/resourceGroups/${RESOURCE_GROUP_EXPECTED}/providers/Azure.ResourceManager.Resources/topLevelTrackedResources/top`,
+        allowReserved: true
+      },
+      "extension"
+    )
+      .get();
+    assert.strictEqual(resourceResult.status, "200");
+    assert.deepEqual(resourceResult.body, validResourceExtensionsResource);
+  });
+
+  it("should createOrUpdate ExtensionsResources ", async () => {
+    const resourceGroupResult = await client
+      .path(
+        "/{resourceUri}/providers/Azure.ResourceManager.Resources/extensionsResources/{extensionsResourceName}",
+        {
+          value: `subscriptions/${SUBSCRIPTION_ID_EXPECTED}/resourceGroups/${RESOURCE_GROUP_EXPECTED}`,
+          allowReserved: true
+        },
+        "extension"
+      )
+      .put({
+        body: {
+          properties: {
+            description: "valid",
+          }
+        }
+      });
+    assert.strictEqual(resourceGroupResult.status, "200");
+    assert.deepEqual(resourceGroupResult.body, validResourceGroupExtensionsResource);
+
+    const subscriptionResult = await client
+      .path(
+        "/{resourceUri}/providers/Azure.ResourceManager.Resources/extensionsResources/{extensionsResourceName}",
+        {
+          value: `subscriptions/${SUBSCRIPTION_ID_EXPECTED}`,
+          allowReserved: true
+        },
+        "extension"
+      )
+      .put({
+        body: {
+          properties: {
+            description: "valid",
+          }
+        }
+      });
+    assert.strictEqual(subscriptionResult.status, "200");
+    assert.deepEqual(subscriptionResult.body, validSubscriptionExtensionsResource);
+
+    const tenantResult = await client
+      .path(
```

Comment:
Could you share the link in test server with me?

Reply:
https://github.com/Azure/typespec-azure/blob/main/packages/azure-http-specs/specs/azure/resource-manager/resources/mockapi.ts